src/plugins/event-plugins/TTIPlugin.ts

```diff
@@ -0,0 +1,45 @@
+import { Plugin } from '../Plugin';
+
+import { TTIBoomerang } from '../../time-to-interactive/TTIBoomerang';
+import { TimeToInteractiveEvent } from '../../events/time-to-interactive-event';
+import { TIME_TO_INTERACTIVE_EVENT_TYPE } from '../utils/constant';
+import { PluginContext } from 'plugins/types';
+import { isLongTaskSupported } from '../../utils/common-utils';
+
+export const TTI_EVENT_PLUGIN_ID = 'time-to-interactive';
+
+export class TTIPlugin implements Plugin {
```

Comment:
Question: by convention, should we be using InternalPlugin?

Reply:
I used plugin as we are planning not to add this into the web bundle itself and rather is just an additional plugin consumers can consume optionally. Therefore, I reasoned it did not meet the criteria of an InternalPlugin. I am open to changing this either way

Reply:
I agree -- this class should extend InteralPlugin.

Because it lives in the aws-rum-web package, it should have the internal prefix to avoid naming conflicts.

Reply:
changed to internal plugin